### PR TITLE
[RA] Resolve most WIN32 defines

### PR DESCRIPTION
**Pull request by hifi**
This also removes all lores code that had WIN32 guards. If someone wants to restore lores mode at some point they need to look at the history.

First step towards portability for RA.

### CODE REVIEW

File: redalert/init.cpp [hunk 1]
```diff
@@ -2008,16 +1924,6 @@ void Init_Random(void)
     CryptRandom.Seed_Bit(t.wDayOfWeek);
     CryptRandom.Seed_Bit(t.wMonth);
     CryptRandom.Seed_Bit(t.wYear);
-#else
```
Comment (tomsons26) [inline redalert/init.cpp:2009]: This would be a valid else case for other systems

File: redalert/menus.cpp [hunk 2]
```diff
@@ -830,15 +791,9 @@ int Main_Menu(unsigned long)
         **	to the cryptographic random number generator.
         */
         if (input != 0) {
-#ifdef WIN32
             SYSTEMTIME t;
             GetSystemTime(&t);
             CryptRandom.Seed_Byte((char)t.wMilliseconds);
-#else
```
Comment (tomsons26) [inline redalert/menus.cpp:837]: This would be a valid else case for other systems

**Comment by tomsons26** [comment]
Touching IPX code is pointless and just pollutes the history, it would go soon anyway.

File: redalert/init.cpp (hunk 1 above)
Comment (hifi) [reply redalert/init.cpp:2009]: It's also trivial.

**Comment by hifi** [comment]
Building currently still depends on IPX shims. Feel free to remove it either before or after this portability cleanup.

File: redalert/session.cpp [hunk 3]
```diff
@@ -783,53 +783,52 @@ void SessionClass::Read_Scenario_Descriptions(void)
...
-            CCFileClass file(name);
-            INIClass ini;
-            ini.Load(file);
-
-            int count = ini.Entry_Count("Missions");
-            for (int index = 0; index < count; index++) {
-                char const* fname = ini.Get_Entry("Missions", index);
-                char buffer[128];
-                ini.Get_String("Missions", fname, "", buffer, sizeof(buffer));
+    **	Fetch any scenario packet lists and apply them first.
```
Comment (tomsons26) [inline redalert/session.cpp:786]: This is Windows specific code

Comment (hifi) [reply redalert/session.cpp:786]: It is. This PR does not try to rewrite Windows specific code.

File: redalert/menus.cpp (hunk 2 above)
Comment (hifi) [reply redalert/menus.cpp:837]: In most cases the Windows version is a better base than the DOS version which is not actively used anymore.

**Comment by tomsons26** [comment]
init.cpp menus.cpp score.cpp session.cpp seem sketchy to me, all others seem fine. Someone else reviewing pay close attention to these.
min/max change might have many underlaying issues, WW templated it for a reason

**Comment by hifi** [comment]
The minmax changes are no-op because they were behind ifndef WIN32. I just moved the implementation we had in function.h to replace it.

File: redalert/init.cpp (hunk 1 above)
Comment (OmniBlade) [reply redalert/init.cpp:2009]: Why not just replace both with a version that works cross platform assuming its not that much work?

**Review by OmniBlade — COMMENTED** [review]
Ideally we strip out all use of the macro/custom template min/max and replace it with the std::min/max from <algorithm>

File: redalert/init.cpp (hunk 1 above)
Comment (hifi) [reply redalert/init.cpp:2009]: That's fair, I can do it in the Linux compat PR if that's okay.

**Review by OmniBlade — APPROVED** [review]
Looks okay to me, RA standalone still builds and runs fine so lets merge?

**Pull request merged**